### PR TITLE
Template spec solution update for Apache Tomcat

### PR DESCRIPTION
**Pull request by vmanojreddy**
Change(s):

See guidance below
Reason for Change(s):
This PR is part of template spec solution migration for solution Tomcat.

See guidance below
Version Updated:
2.0.0

See guidance below
Testing Completed:
The link for test results is
https://microsoftapc.sharepoint.com/:w:/r/teams/SentinelCATTeam/_layouts/15/Doc.aspx?sourcedoc=%7B39D6C801-E9DF-4A23-9AF1-411F89A7A4DC%7D&file=Sentinel%20Solution%20Migration%20-TemplateSpec%20-%20Tomcat.docx&action=default&mobileredirect=true&share=IQEByNY53-kjSprxQR-Jp6TcAbuMhYs9Szeo3EivrQpICsk

### CODE REVIEW

File: Solutions/Tomcat/Data/Solution_Tomcat.json [hunk 1]
```diff
@@ -1,14 +1,45 @@
...
   ],
   "Parsers": [
     "Parsers/TomcatEvent.txt"
   ],
-  "Metadata": "SolutionMetadata.json",
+  "Workbooks": [ 
+    "Workbooks/Tomcat.json" 
+   ],
+  "Analytic Rules": [
+    "Analytic Rules/TomcatCommandsinRequest.yaml",
```
Comment (v-amolpatil) [inline Solutions/Tomcat/Data/Solution_Tomcat.json:16]: File names are repeated. Please update file names.

File: Solutions/Tomcat/Package/mainTemplate.json [hunk 2]
```diff
@@ -25,29 +25,416 @@
...
+              "apiVersion": "2021-03-01-preview",
+              "type": "Microsoft.OperationalInsights/workspaces/providers/dataConnectors",
+              "location": "[parameters('workspace-location')]",
+              "kind": "GenericUI",
+              "properties": {
+                "connectorUiConfig": {
+                  "id": "[variables('_uiConfigId1')]",
+                  "title": "Apache Tomcat",
+                  "publisher": "Apache",
+                  "descriptionMarkdown": "The Apache Tomcat data connector provides the capability to ingest [Apache Tomcat](http://tomcat.apache.org/) events into Azure Sentinel. Refer to [Apache Tomcat documentation](http://tomcat.apache.org/tomcat-10.0-doc/logging.html) for more information.",
```
Comment (v-amolpatil) [inline Solutions/Tomcat/Package/mainTemplate.json:219]: Please rename `Azure Sentinel` to `Microsoft Sentinel`. There are 2 occurences in this file at line 219 and 446.

File: Solutions/Tomcat/Package/mainTemplate.json [hunk 3]
```diff
@@ -25,29 +25,416 @@
...
-    "connector1-source": "[concat('/subscriptions/',subscription().subscriptionId,'/resourceGroups/',resourceGroup().name,'/providers/Microsoft.OperationalInsights/workspaces/',parameters('workspace'),'/providers/Microsoft.SecurityInsights/dataConnectors/',parameters('connector1-name'))]",
-    "_connector1-source": "[variables('connector1-source')]",
-    "ApacheTomcatConnector": "ApacheTomcatConnector",
-    "_ApacheTomcatConnector": "[variables('ApacheTomcatConnector')]",
-    "workspace-dependency": "[concat('Microsoft.OperationalInsights/workspaces/', parameters('workspace'))]",
-    "TomcatEvent_Parser": "TomcatEvent_Parser",
-    "_TomcatEvent_Parser": "[variables('TomcatEvent_Parser')]",
-    "sourceId": "azuresentinel.azure-sentinel-solution-apachetomcat",
-    "_sourceId": "[variables('sourceId')]"
+    "solutionId": "azuresentinel.azure-sentinel-solution-apachetomcat-preview",
```
Comment (v-amolpatil) [inline Solutions/Tomcat/Package/mainTemplate.json:41]: please remove `-preview` from `solutionId` attributes value.

File: Solutions/Tomcat/Data Connectors/Connector_Tomcat_agent.json [hunk 4]
```diff
@@ -2,7 +2,7 @@
     "id": "ApacheTomcat",
     "title": "Apache Tomcat",
     "publisher": "Apache",
-    "descriptionMarkdown": "The Apache Tomcat data connector provides the capability to ingest [Apache Tomcat](http://tomcat.apache.org/) events into Azure Sentinel. Refer to [Apache Tomcat documentation](http://tomcat.apache.org/tomcat-10.0-doc/logging.html) for more information.",
+    "descriptionMarkdown": "The Apache Tomcat data connector provides the capability to ingest [Apache Tomcat](http://tomcat.apache.org/) events into Microsoft Sentinel. Refer to [Apache Tomcat documentation](http://tomcat.apache.org/tomcat-10.0-doc/logging.html) for more information.",
```
Comment (v-amolpatil) [inline Solutions/Tomcat/Data Connectors/Connector_Tomcat_agent.json:5]: Please check if we need to replace `data connector` text with `solution`
image

File: Solutions/Tomcat/Data/Solution_Tomcat.json [hunk 5]
```diff
@@ -1,14 +1,45 @@
...
+  "Name": "Tomcat",
+  "Author": "Microsoft - support@microsoft.com",
+  "Logo": "<img src=\"https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/Logos/Azure_Sentinel.svg\" width=\"75px\" height=\"75px\">",
+  "Description": "The Apache Tomcat data connector provides the capability to ingest [Apache Tomcat](http://tomcat.apache.org/) events into Microsoft Sentinel. Refer to [Apache Tomcat documentation](http://tomcat.apache.org/tomcat-10.0-doc/logging.html) for more information.\r\n \r\n**Underlying Microsoft Technologies used:** \r\n \r\n This solution takes a dependency on the following technologies, and some of these dependencies either may be in [Preview](https://azure.microsoft.com/support/legal/preview-supplemental-terms/) state or might result in additional ingestion or operational costs:\r\n \r\na. [Azure Monitor HTTP Data Collector API](https://docs.microsoft.com/azure/azure-monitor/logs/data-collector-api)",
```
Comment (v-amolpatil) [inline Solutions/Tomcat/Data/Solution_Tomcat.json:5]: Replace text `data connector` to `solution` as per text change file shown below:

image

File: Solutions/Tomcat/Package/createUiDefinition.json [hunk 6]
```diff
@@ -60,41 +60,405 @@
...
+          "preValidation": "Configure the workbooks",
+          "postValidation": "Done"
+        },
+        "bladeTitle": "Workbooks",
+        "elements": [
+          {
+            "name": "workbooks-text",
+            "type": "Microsoft.Common.TextBlock",
+            "options": {
+              "text": "This solution installs workbook to help you gain insights into the telemetry collected in Microsoft Sentinel. After installing the solution, start using the workbook in Manage solution view."
```
Comment (v-amolpatil) [inline Solutions/Tomcat/Package/createUiDefinition.json:98]: Please check if `workbook` has to be updated to `workbook(s)` as per text change document.

image

File: Solutions/Tomcat/Data/Solution_Tomcat.json (hunk 5 above)
Comment (vmanojreddy) [reply Solutions/Tomcat/Data/Solution_Tomcat.json]: Done

**Review by v-amolpatil — APPROVED** [review]

**Pull request merged**